### DISCUSSION TITLE
homebrew and fish shell

### DISCUSSION THREAD

**Discussion by thistlillo** [fish-shell/fish-shell#9551] [General]
I have curiosity about the fish shell and homebrew. I asked this on the homebrew discussions but did not receive a definite answer.

I downloaded the Mac OS from the website and it is ~30MB in size.

Then I installed the (fish) shell using homebrew: it downloaded plenty of libraries and other support software. For example, openjdk (that was not installed). Openjdk is more than 300MB in size. However, if I look at the dependencies here https://github.com/fish-shell/fish-shell, it seems to rely upon a very limited set of libraries.

So question: who is packaging the cask "fish-shell"? How many MBs does this shell occupy? 

If I had a normal-size HD I would not probably care, but with a 256GB MacBook Air, hundreds of MBs for the fish shell is quite a luxury I cannot afford.

**Comment by ridiculousfish** [comment]
fish-shell does not depend on OpenJDK. What command did you run exactly?

The fish-shell Homebrew [recipe is here](https://github.com/Homebrew/homebrew-core/blob/fea3233bc2c2b9cf0b312c9ee42302d461a07599/Formula/fish.rb). You'll see it depends on cmake, ncurses, and pcre2; but not OpenJDK.

**Reply by faho** [reply]
I'm assuming they installed "[glassfish](https://formulae.brew.sh/formula/glassfish)" instead, which is indeed a java application.

**Reply by thistlillo** [reply]
No no, I installed `fish`: `brew install fish`

May be it is related to this: [https://github.com/Homebrew/discussions/discussions/57](https://github.com/Homebrew/discussions/discussions/57) and new dependencies were added to packages that got upgraded. 

I use homebrew often, but I have never "studied" its working. It is the first time brew downloaded all that stuff when installing a simple package.

**Reply by faho** [reply]
Check the actual [fish formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/fish.rb), it does not have an openjdk dependency.

If openjdk got added onto your system, it's not because of that.

>May be it is related to this: https://github.com/orgs/Homebrew/discussions/57 and new dependencies were added to packages that got upgraded.

It is possible that another package you use gained an openjdk dependency. Fish doesn't have one, the main fish formula that you would typically get with `brew install fish` doesn't have one and I can't find a separate fish formula/cask/tap/... that has one, and it wouldn't make any sense because nothing about fish is written in java.

**Reply by faho** [reply]
From your discussion, it seems you can run `brew deps fish` to see what the fish you installed depends on, and `brew uses openjdk` to see what depends on openjdk.

**Reply by thistlillo** [reply]
> From your discussion, it seems you can run `brew deps fish` to see what the fish you installed depends on, and `brew uses openjdk` to see what depends on openjdk.

I will check it as soon as I get home tonight, where my Mac is, I cannot do it now.

**Reply by ridiculousfish** [reply]
When you run `brew install foo` it will sometimes run `brew update --auto-update` which can install a bunch of other stuff unrelated to foo. That's my guess as to what happened.

**Comment by thistlillo** [comment]
I checked the dependencies: as expected.
```
base ❯ brew deps fish 
ncurses
pcre2
```
both without further dependencies.

So may be it ran a `--auto-update`.

Thanks for your support.
